### PR TITLE
refactor(client): Replace p-promise with DOM promises

### PR DESCRIPTION
**Pull request by shane-tomlinson**
ref https://github.com/mozilla/fxa-js-client/pull/263

### CODE REVIEW

**Comment by vladikoff** [comment]
(I'm a bit worried about IE11 in this case after looking at GA stats. However if we improve our email link verification (server verified?) or "Update your browser messaging" I would feel better)

**Comment by shane-tomlinson** [comment]
> (I'm a bit worried about IE11 in this case after looking at GA stats. However if we improve our email link verification (server verified?) or "Update your browser messaging" I would feel better)

I have a plan for that. We can either use IE conditional loading for a polyfill, or https://philipwalton.com/articles/deploying-es2015-code-in-production-today/.

**Comment by vladikoff** [comment]
Yaaay!

**Comment by shane-tomlinson** [comment]
@vladikoff - I'm calling this ready!

I included the polyfill in the bundle for a couple of reasons.

1. We have a ton of people verifying in IE and Android pre 4.5 where Promises were introduced. We still want to support them.
2. Adding the script to a `nomodule` polyfill bundle would be nice, except Firefox still requests the resources so there's no real benefit.

The good news is, the polyfill is slightly smaller than p.

**Comment by shane-tomlinson** [comment]
I'm trying to get an fxa-dev box setup to test on IE. es6-promise.dev.lcip.org, unfortunately it keeps failing to connect to the ELB.

**Comment by shane-tomlinson** [comment]
@mozilla/fxa-devs - r? (at least in principle until I can get the es6-promise set up to test)

**Comment by vladikoff** [comment]
Let's use the latest js-client (with native promises) in this PR

**Comment by shane-tomlinson** [comment]
This is running on https://es6-promise.dev.lcip.org and works great with IE10. Ready for review.

@mozilla/fxa-devs - r?

File: app/scripts/lib/promise.js [hunk 1]
```diff
@@ -5,26 +5,45 @@
...
+
+  module.exports = {
+    /**
+     * Return a promise that resolves after `delayMS`.
+     *
+     * @param {Number} delayMS
+     * @returns {Promise}
+     */
+    delay (delayMS) {
+      return new Promise((resolve, reject) => {
```
[the remark this answers is not in the document]
Comment (shane-tomlinson) [reply app/scripts/lib/promise.js]: I was wondering that too, I should probably ditch reject if it's not used.

**Pull request merged**